challenge-14/challenge-14.js

```diff
@@ -1,97 +1,147 @@
-console.
-*/
-console.log( '\nOperation 2:' );
-// ?
-
-/*
-Crie um array chamado `name`. Cada elemento desse array deve ser uma sílaba
-do seu nome. Vamos reduzir esse array, juntando todas as sílabas, mas usando
-a "língua do P".
-PS.: Lembra da língua do "P"? Não? A língua do "P" é uma brincadeira
-infantil, onde você coloca a letra "P" antes de cada sílaba de uma palavra
-falada, como se você estivesse falando em código xD
-*/
-console.log( '\nSeu nome na língua do "P":' );
-// ?
-
-/*
-Crie uma variável chamada `inversedName`, que reduzirá o array em uma string
-e atribuirá o seu nome invertido (usando o array criado acima).
-*/
-console.log( '\nInversed Name:' );
-// ?
-
-/*
-Mostre no console o array `numberObjects`.
-*/
-console.log( '\nNumber objects' );
-// ?
-
-/*
-Verifique se existem em algum índice de numberObjects um objeto ìgual a
-{ number: 2 }. Se houver, mostre no console:
-- "Existe um objeto { number: 2 } em numberObjects!"
-Senão, mostre a frase:
-- "Não existe um objeto { number: 2 } em numberObjects :("
-Consegue prever o resultado? Deixe uma mensagem no console tentando explicar
-o que acontece ;)
-*/
-console.log( '\nExiste um { number: 2 } em numberObjects?' );
-// ?
-
-/*
-Fazendo o mesmo do exercício acima, mas começando a buscar do último índice,
-será que obtemos um resultado diferente? Faça a busca a partir do índice 2.
-*/
-console.log( '\nE buscando a partir do último índice, o { number: 2 } existe?' );
-// ?
-
-/*
-Verifique se `justMod2Or3` é um array. Se for, mostre-o no console, no
-formato de String.
-*/
-console.log( '\njustMod2Or3 é um array? Se for, a representação dele em String é:' );
-// ?
+(function() {
+  /*
+  Envolva todo o código desse desafio em uma IIFE.
+  */
+
+  /*
+  Crie um array chamado numberObjects. Esse array deve ter 10 elementos. Cada
+  elemento será um objeto no formato:
+  { number: [NUMBER] }
+  Os números devem ser de 1 a 10.
+  Mostre esse array no console.
+  */
+  console.log( 'Number Objects Array:' );
+  var numberObjects = [];
+
+  for( var counter = 1; counter <= 10; counter++ ) {
+    numberObjects.push( { number: counter } );
+  }
+
+  console.log( numberObjects );
+
+  /*
+  Crie um array chamado `justNumbers`, que terá como elementos somente os
+  números do array criado acima. Mostre esse novo array no console.
+  */
+  console.log( '\nJust Numbers:' );
+  var justNumbers = numberObjects.map( function( item, index ) {
+    for( var property in numberObjects[index] ) {
+      return item[ property ];
+    }
+  } );
+
+  console.log( justNumbers );
+
+  /*
+  Crie um novo array chamado `justMod2Or3`, que receberá do array criado acima
+  somente os números que forem divisíveis por 2 ou 3. Mostre esse novo array
+  no console.
+  */
+  console.log( '\nJust module of division by 2 or 3:' );
+  var justMod2Or3 = justNumbers.filter( function( item ) {
+    if( item % 2 === 0 || item % 3 === 0  ) {
+      return item;
```

Comment:
-1
O `filter` por padrão já retorna o `item` atual, se o returno desse método for verdadeiro. Então aqui, você não precisa de um `if`, somente precisa retornar o teste para ver se o item é divisível por 2 ou 3 ;)